src/compiler/parser.ts

```diff
@@ -6673,11 +6673,16 @@ namespace ts {
         }
 
         function parseFunctionBlockOrSemicolon(flags: SignatureFlags, diagnosticMessage?: DiagnosticMessage): Block | undefined {
-            if (token() !== SyntaxKind.OpenBraceToken && canParseSemicolon()) {
-                parseSemicolon();
-                return;
+            if (token() !== SyntaxKind.OpenBraceToken) {
+                if (flags & SignatureFlags.Type) {
+                    parseTypeMemberSemicolon();

```

Comment:
does this change parsing of other function-like declarations, particularly methods? Or, to reverse the question: how did method signature parsing work before? Does it not use parseFunctionBlockOrSemicolon? If not, why does parseAccessDeclaration?

Reply:
`parseMethodDeclaration` calls `parseFunctionBlockOrSemicolon`, if that's what you mean.

Reply:
`parseTypeMember` doesn't parse `,`, `;` (`parseTypeMemberSemicolon`) tokens after accessors

https://github.com/microsoft/TypeScript/blob/569cdf1b074dd7462c53102e0f69bfc9863a12d7/src/compiler/parser.ts#L3633-L3635

`parseAccessorDeclaration` is general method for type members/literals/class members for. Parsing `,`, `;` should be allowed for type members.`parseFunctionBlockOrSemicolon` has signature flags that include the `Type` option. I used it to allow parsing `,`, `;`. In other cases, the behavior stays the same. (`parseFunctionBlockOrSemicolon(SignatureFlags.None)`) Does that make sense?